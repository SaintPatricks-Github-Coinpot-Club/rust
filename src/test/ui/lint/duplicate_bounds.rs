#![deny(duplicate_bounds)]

trait DupDirectAndWhere {}
fn dup_direct_and_where<T: DupDirectAndWhere>(t: T)
where
    T: DupDirectAndWhere,
    //~^ ERROR this trait bound has already been specified
    T: DupDirectAndWhere,
    //~^ ERROR this trait bound has already been specified
{
    unimplemented!();
}

trait DupDirect {}
fn dup_direct<T: DupDirect + DupDirect>(t: T) {
    //~^ ERROR this trait bound has already been specified
    unimplemented!();
}

trait DupWhere {}
fn dup_where<T>(t: T)
where
    T: DupWhere + DupWhere,
    //~^ ERROR this trait bound has already been specified
{
    unimplemented!();
}

trait NotDup {}
fn not_dup<T: NotDup, U: NotDup>((t, u): (T, U)) {
    unimplemented!();
}

fn dup_lifetimes<'a, 'b: 'a + 'a>() {}
//~^ ERROR this lifetime bound has already been specified

fn dup_lifetimes_generic<'a, T: 'a + 'a>() {}
//~^ ERROR this lifetime bound has already been specified

fn dup_lifetimes_where<T: 'static>()
where
    T: 'static,
    //~^ ERROR this lifetime bound has already been specified
{
}

trait Everything {}
fn everything<T: Everything + Everything, U: Everything + Everything>((t, u): (T, U))
//~^ ERROR this trait bound has already been specified
//~| ERROR this trait bound has already been specified
where
    T: Everything + Everything + Everything,
    //~^ ERROR this trait bound has already been specified
    //~| ERROR this trait bound has already been specified
    //~| ERROR this trait bound has already been specified
    U: Everything,
    //~^ ERROR this trait bound has already been specified
{
    unimplemented!();
}

fn main() {}
