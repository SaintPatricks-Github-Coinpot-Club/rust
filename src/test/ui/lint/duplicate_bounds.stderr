error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:6:8
   |
LL |     T: DupDirectAndWhere,
   |        ^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/duplicate_bounds.rs:1:9
   |
LL | #![deny(duplicate_bounds)]
   |         ^^^^^^^^^^^^^^^^
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:8:8
   |
LL |     T: DupDirectAndWhere,
   |        ^^^^^^^^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:15:30
   |
LL | fn dup_direct<T: DupDirect + DupDirect>(t: T) {
   |                              ^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:23:19
   |
LL |     T: DupWhere + DupWhere,
   |                   ^^^^^^^^
   |
   = help: consider removing this trait bound

error: this lifetime bound has already been specified
  --> $DIR/duplicate_bounds.rs:34:31
   |
LL | fn dup_lifetimes<'a, 'b: 'a + 'a>() {}
   |                               ^^
   |
   = help: consider removing this lifetime bound

error: this lifetime bound has already been specified
  --> $DIR/duplicate_bounds.rs:37:38
   |
LL | fn dup_lifetimes_generic<'a, T: 'a + 'a>() {}
   |                                      ^^
   |
   = help: consider removing this lifetime bound

error: this lifetime bound has already been specified
  --> $DIR/duplicate_bounds.rs:42:8
   |
LL |     T: 'static,
   |        ^^^^^^^
   |
   = help: consider removing this lifetime bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:48:31
   |
LL | fn everything<T: Everything + Everything, U: Everything + Everything>((t, u): (T, U))
   |                               ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:48:59
   |
LL | fn everything<T: Everything + Everything, U: Everything + Everything>((t, u): (T, U))
   |                                                           ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:52:8
   |
LL |     T: Everything + Everything + Everything,
   |        ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:52:21
   |
LL |     T: Everything + Everything + Everything,
   |                     ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:52:34
   |
LL |     T: Everything + Everything + Everything,
   |                                  ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound has already been specified
  --> $DIR/duplicate_bounds.rs:56:8
   |
LL |     U: Everything,
   |        ^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: aborting due to 13 previous errors

